core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java

```diff
@@ -1051,6 +1051,69 @@
     sql(mv, query).ok();
   }
 
+  @Test void testAggCallArgExpressedByMvProjs1ForFail() {
```

Comment:
You are fixing two mv recognition cases in one pr, 'max' and 'count(1) or count(*)' in query.
But actually they are not related, can you split this into two prs?

Reply:
Ok, I'll split it into two pr.
Thanks for your review.